core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -308,6 +308,11 @@ public static boolean isConflictException(Throwable t) {
         ShardRouting routingEntry();
 
         /**
+         * primary term for this primary (fixed when primary shard is created)
+         */
+        long primaryTerm();
```

Comment:
conversely, we could as well use the primaryTerm on the replicaRequest when failing the shard. This means that we don't need to extend the `Primary` interface with `primaryTerm`.